cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/BusinessMetadataDataset.java

```diff
@@ -129,23 +198,49 @@ public BusinessMetadataRecord getBusinessMetadata(Id.NamespacedId targetId, Stri
   }
 
   /**
+   * Retrieves all the properties for the specified {@link Id.NamespacedId}.
+   *
+   * @param targetId the {@link Id.NamespacedId} for which properties are to be retrieved
+   * @return the properties of the specified {@link Id.NamespacedId}
+   */
+  public Map<String, String> getProperties(Id.NamespacedId targetId) {
```

Comment:
I agree its more flexible. However, we cannot return ``BusinessMetadataRecord`` as is from the REST API. So was wondering that we will have to do a pass over the ``List<BusinessMetadataRecord>`` passed from here, and convert it into a representation suitable for REST somewhere downstream.

Since this is an internal API, I'm going to keep it as is, maybe in future we can return BusinessMetadataRecord, and have a custom serializer to omit the fields that are not required from the REST API.